graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1949,35 +2053,81 @@ def _create_split_files(
         The metadata dict elements for the train/test/val masks, to be added to the caller's
         edge/node type metadata.
         """
-        # If the user did not provide a split rate we use a default
+
+        # Use custom column names if requested
+        if mask_field_names:
+            mask_names = mask_field_names
+        else:
+            mask_names = ("train_mask", "val_mask", "test_mask")
+
+        # TODO: Make this an argument to the write functions to not rely on function scope?
         split_metadata = {}
+
+        def write_masks_numpy(np_mask_arrays: Sequence[np.ndarray]):
```

Comment:
What about put these two functions into utils.py?

Reply:
My rule of thumb to make a function public is : "is this function used in two place in the codebase?". 

e.g. that's why I moved `_create_metadata_entry` from an inner function to be part DGHL in this PR.

Right now the answer is no for these functions, if we see a need to re-use them we can pull them out.